pkg/proxy/iptables/proxier.go

```diff
@@ -166,12 +166,30 @@ type endpointsInfo struct {
 	chainName utiliptables.Chain
 }
 
+// Returns just the IP part of an IP or IP:port or endpoint string. If the IP
+// part is an IPv6 address enclosed in brackets (e.g. "[fd00:1::5]:9999"),
+// then the brackets are stripped as well.
+func ipPart(s string) string {
+	if ip := net.ParseIP(s); ip != nil {
+		// IP address without brackets
+		return s
+	}
+	if s[0] == '[' && s[len(s)-1] == ']' {
```

Comment:
is this a legit notation?  Why wouldn't `ParseIP` handle it correctly?

Reply:
You're right, this is not legit. The brackets should only appear for IPv6:port or when IPv6 address appears in a URL. I'll delete these lines.